### ISSUE TITLE
Julia 1.0+ not supported

### ISSUE THREAD

**Issue by alexjbuck** [JuliaMPC/NLOptControl.jl#24]
This is a great tool and I would like to work on porting this over to the current Julia version, at least 1.0+. I'm new to open source contribution, so hopefully this is the correct pathway. I have created a fork to begin the required changes.

**Comment by huckl3b3rry87** [comment]
Awesome 👍 thanks!

**Comment by huckl3b3rry87** [comment]
I will help with this effort when I can, it it exciting to have another
developer.

**Comment by alexjbuck** [comment]
Awesome! I wasn't sure if you were still using/supporting this or not, it had been a few months since the last commit.

**Comment by huckl3b3rry87** [comment]
Yeah, I still use it, but without support from the community I don't have
time to update for julia 1.0. But I would like to see it maintained and
will help in my spare time.

**Comment by mfalt** [comment]
It would be great if this is upgraded to 1.0. If you have any problems with the PR or any updates that are unclear, I am happy to help you resolve them. I think this package could be very useful. We are already using it as part of a toolchain for a student-lab at our university.

The only thing that I think should be added is something along the lines of https://github.com/JuliaMPC/NLOptControl.jl/issues/5 , I am looking a bit at how to use ModelingToolkit.jl for this.

**Comment by azev77** [comment]
@huckl3b3rry87 @alexjbuck @mfalt 
Guys I just tried `Pkg.add("NLOptControl")` following the doc & unable to install. 

I'm writing a note on optimal control for econ & would be awesome to show off your package. 
Any idea on a time-line? 

FYI:
julia> versioninfo()
Julia Version 1.3.0
Commit 46ce4d7933 (2019-11-26 06:09 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Xeon(R) CPU E5-1630 v3 @ 3.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, haswell)
Environment:
  JULIA_EDITOR = "C:\Users\azevelev\AppData\Local\atom\app-1.43.0\atom.exe"  -a
  JULIA_NUM_THREADS = 4

**Comment by mfalt** [comment]
@azev77 I created a PR (https://github.com/JuliaMPC/NLOptControl.jl/pull/25) that updates to julia 1, but I wans't able to figure out what is going wrong. Since I didn't write the original code, I'm not able to understand all parts. I think we have to wait on @huckl3b3rry87 to see if he can fix the last parts.

**Comment by huckl3b3rry87** [comment]
@azev77 you can use it with [julia 0.6.4](https://travis-ci.org/JuliaMPC/NLOptControl.jl), is there a reason you need the latest version of julia?

@mfalt sorry for the delay on this, I will try to look into this soon

**Comment by azev77** [comment]
1 Yeah, I need it for the latest Julia. 
2 I'd be happy to contribute a few examples from economics to your docs once it works, if you're interested

**Comment by huckl3b3rry87** [comment]
@azev77 sounds good, yeah extra examples are always great

What is your timeline?

**Comment by azev77** [comment]
I'm preparing a blog post on Julia for economics.
I wanna write it over the next two weeks. (A lot of it is already done.)

I can probably post my current version w/o your package, & update it when your package works.

**Comment by huckl3b3rry87** [comment]
@azev77 very cool, I am interested in seeing the application of NLOptControl to economics. Are you designing an algorithm for options trading?

I will try to look at it this weekend

**Comment by azev77** [comment]
That would be awesome. 
I'm writing a post on the advantages of Julia for an Econ audience. 

The first example is a routine consumption-saving model every econ student solves (by hand) in grad school. 
-then I spice it up w/ borrowing constraints 
(currently I do all this w/ DifferentialEquations.jl, but the NLOptControl interface is sweet!) 
-then I add endogenous labor & housing choice (with adjustment costs) 

The second example is firm profit maximization. 
Again, starting w/ simplest model, then adding layers. 

The third example is the optimal-growth model. 

To solve finance problems (such as options trading etc) I need some of the state variables to be stochastic. This is usually done by solving an HJB equation. [EconPDEs](https://github.com/matthieugomez/EconPDEs.jl) is amazing for certain problems, but it doesn't handle boundaries well yet.

**Comment by azev77** [comment]
@huckl3b3rry87 any luck with this?

**Comment by roiholtzman** [comment]
I have Julia 1.3. Do you think I can install Julia 0.6.4 on my machine as well without conflicting with the 1.3 version I have? (I read somewhere that maybe for versions below 0.7 this can be an issue).

**Comment by huckl3b3rry87** [comment]
It should be fine

**Comment by huckl3b3rry87** [comment]
this should do it https://github.com/JuliaMPC/NLOptControl.jl/pull/30
